src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -0,0 +1,246 @@
+import { BusinessLookupServices, LegalServices } from '@/services'
+import { BusinessLookup } from '@bcrs-shared-components/business-lookup'
+import { BusinessIF, BusinessLookupIF, EmptyBusinessLookup } from '@/interfaces'
+
+@Component({
+  components: {
+    BusinessLookup
+  }
+})
+export default class AmalgamatingBusinesses extends Mixins(CommonMixin) {
+  @Getter(useStore) getAmalgamatingBusinesses!: Array<BusinessIF>
+  @Getter(useStore) isRoleStaff!: boolean
+
+  @Action(useStore) setAmalgamatingBusinesses!: (x: Array<BusinessIF>) => void
+
+  // Local properties
+  amalgamatingBusinessesValid = false
+  amalgamatingBuinesses = []
+  initialBusinessLookupObject = EmptyBusinessLookup
+
+  // Add an Amalgamating Business button properties
+  isAddingAmalgamatingBusiness = false
+  addAmalgatingBusinessDisabled = false
+
+  // Add an Amalgamating Foreign Business button properties
+  isAddingAmalgamatingForeignBusiness = false
+  addAmalgatingForeignBusinessDisabled = false
+
+  readonly BusinessLookupServices = BusinessLookupServices
+
+  // Cancel button in "Add an Amalgamating Business" is pressed.
+  addAmalgamatingBusinessCancel (): void {
+    this.isAddingAmalgamatingBusiness = false
+    // Enable buttons
+    this.addAmalgatingBusinessDisabled = false
+    this.addAmalgatingForeignBusinessDisabled = false
+  }
+
+  // "Add an Amalgamating Business" button is pressed.
+  onAddBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = true
+    this.isAddingAmalgamatingForeignBusiness = false
+    // Disable buttons
+    this.addAmalgatingBusinessDisabled = true
+    this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  // "Add an Amalgamating Foreign Business" button is pressed.
+  onAddForeignBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = false
+    this.isAddingAmalgamatingForeignBusiness = true
+    // Disable buttons (Please comment out the lines below or remove in part 2)
+    // this.addAmalgatingBusinessDisabled = true
+    // this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  async setAmalgamatingBusiness (businessLookup: BusinessLookupIF): Promise<void> {
+    // Get the amalgamating business information
+    // Will have a different format depending on the business
+    let business = await LegalServices.fetchBusinessInfo(businessLookup.identifier)
+      .then((response) => {
+        return response?.data?.business
+      }).catch(() => {
+        return businessLookup
+      })
+
+    // Get the address of the amalgamating business
+    if (businessLookup.identifier && business.foundingDate) {
+      const addresses = await LegalServices.fetchAddresses(businessLookup.identifier)
+        .then((data) => {
+          // SP and GP have businessOffice instead of registeredOffice
+          return data?.registeredOffice || data?.businessOffice
+        }).catch(() => {
+          return undefined
+        })
+      if (addresses) {
+        business.officeAddress = addresses
+      }
```

Comment:
Some businesses don't have an address (this is also showcased in the UXPin). If we were able to fetch the address, put it in the business object.

Reply:
I think all businesses have an address. What are you referring to?

Reply:
> Sev, is this a bad thing (2 calls in this function)? Do you think that'll be bad performance?

We knew this coming in. In fact, I think you also need to make an Auth call to fetch the contact info email address.

However, instead of doing all 3 network calls sequentially, you can do them in parallel as we do [here](https://github.com/bcgov/business-filings-ui/blob/5d744948b7e52a69bdafb1ca2e4cd5a14bef5f40/src/App.vue#L483) or in Edit UI.